### PR TITLE
Fix edge case where dangling end merging creates cycles

### PR DESCRIPTION
**Pull request by davidbenjamin**
@meganshand This fixes the Sarah's recent bug, and just makes sense in general.

I am not adding a regression test because our work on linked de Bruijn graphs is going to moot stuff like this soon.

### CODE REVIEW

**Comment by sarahcalvo** [comment]
Thanks @davidbenjamin!  Please let me know when the merge is complete and we will rerun gnomAD analysis.

File: src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java [hunk 1]
```diff
@@ -434,7 +434,12 @@ private AssemblyResult createGraph(final Iterable<GATKRead> reads,
             return null;
         }
 
-        return getAssemblyResult(refHaplotype, kmerSize, rtgraph, aligner);
+        final AssemblyResult result = getAssemblyResult(refHaplotype, kmerSize, rtgraph, aligner);
+        // check whether recovering dangling ends created cycles
+        if (recoverAllDanglingBranches && rtgraph.hasCycles()) {
```
Comment (meganshand) [inline src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java]: It doesn't actually matter if `recoverAllDanglingBranches` right? You want to return null if there are cycles regardless?

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java]: You're right but I'm being conservative about runtime here.  This is all going to be overhauled soon anyway.

Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java]: And don't worry about the code coverage.  It's effectively a random number generator.

Comment (meganshand) [reply src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java]: Sounds good, merge away!

**Pull request merged**